routes/adjustmentMethods/EToAdjustmentMethod.ts

```diff
@@ -0,0 +1,224 @@
+ * @param speed The wind speed measured at the specified height (in miles per hour).
+ * @param height The height of the measurement (in feet).
+ * @returns The approximate wind speed at 2 meters (in miles per hour).
+ */
+export function standardizeWindSpeed( speed: number, height: number ) {
+	return speed * 4.87 / Math.log( 67.8 * height / 3.281 - 5.42 );
+}
+
+// The time at which the formula for clear sky isolation will start/stop yielding a non-negative result.
+SunCalc.addTime( Math.asin( 30 / 990 ) * 180 / Math.PI, "radiationStart", "radiationEnd" );
+
+/**
+ * Approximates total solar radiation for a day given cloud coverage information using a formula from
+ * http://www.shodor.org/os411/courses/_master/tools/calculators/solarrad/
+ * @param cloudCoverInfo Information about the cloud coverage for several periods that span the entire day.
+ * @param coordinates The coordinates of the location the data is from.
+ * @return The total solar radiation for the day (in megajoules per square meter per day).
+ */
+export function approximateSolarRadiation(cloudCoverInfo: CloudCoverInfo[], coordinates: GeoCoordinates ): number {
+	return cloudCoverInfo.reduce( ( total, window: CloudCoverInfo ) => {
+		const radiationStart: moment.Moment = moment( SunCalc.getTimes( window.endTime.toDate(), coordinates[ 0 ], coordinates[ 1 ])[ "radiationStart" ] );
+		const radiationEnd: moment.Moment = moment( SunCalc.getTimes( window.startTime.toDate(), coordinates[ 0 ], coordinates[ 1 ])[ "radiationEnd" ] );
+
+		// Clamp the start and end times of the window within time when the sun was emitting significant radiation.
+		const startTime: moment.Moment = radiationStart.isAfter( window.startTime ) ? radiationStart : window.startTime;
+		const endTime: moment.Moment = radiationEnd.isBefore( window.endTime ) ? radiationEnd: window.endTime;
+
+		// The length of the window that will actually be used (in hours).
+		const windowLength = ( endTime.unix() - startTime.unix() ) / 60 / 60;
+
+		// Skip the window if there is no significant radiation during the time period.
+		if ( windowLength <= 0 ) {
+			return total;
+		}
+
+		const startPosition = SunCalc.getPosition( startTime.toDate(), coordinates[ 0 ], coordinates[ 1 ] );
+		const endPosition = SunCalc.getPosition( endTime.toDate(), coordinates[ 0 ], coordinates[ 1 ] );
+		const solarElevationAngle = ( startPosition.altitude + endPosition.altitude ) / 2;
+
+		// Calculate radiation and convert from watts to megajoules.
+		const clearSkyIsolation = ( 990 * Math.sin( solarElevationAngle ) - 30 ) * 0.0036 * windowLength;
+
+		return total + clearSkyIsolation * ( 1 - 0.75 * Math.pow( window.cloudCover, 3.4 ) );
+	}, 0 );
+}
+
+export interface EToScalingAdjustmentOptions extends AdjustmentOptions {
+	/** The watering site's height above sea level (in feet). */
+	elevation?: number;
+	/** Baseline potential ETo (in inches per day). */
+	baseETo?: number;
```

Comment:
I'm working on a function that will automatically determine baseline ETo using data from [MOD16](https://www.ntsg.umt.edu/project/modis/mod16.php). Once I get that setup, users will able to automatically detect the baseline ETo in the app when configuring the adjustment method.

Reply:
I only had a quick scan of the linked website but I couldn't see how to get their data for, say, London. Am I missing something?

Reply:
The data is available in a massive [43200x16800 pixel grayscale map](http://files.ntsg.umt.edu/data/NTSG_Products/MOD16/MOD16A3.105_MERRAGMAO/Geotiff/MOD16A3_PET_2000_to_2013_mean.tif) where the value of each pixel represents the average annual potential ETo for that location. I'm working on a script that fills in missing data (cities don't have any data associated with them since they have non-vegetated land cover, so I've extrapolated data by averaging the value of surrounding pixels), and saves the data in an uncompressed binary format so we can easily map geo-coordinates to an offset in the file. We'll be able to store the file on disk (and not need to worry about loading it into memory) and quickly access the baseline ETo for a given location by seeking to the calculated offset.

Reply:
Best of luck 8)